src/main/java/io/cryostat/net/web/http/api/beta/RecordingGetHandler.java

```diff
@@ -0,0 +1,134 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package io.cryostat.net.web.http.api.beta;
+
+import java.nio.file.Path;
+import java.util.EnumSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.ExecutionException;
+
+import javax.inject.Inject;
+
+import io.cryostat.net.AuthManager;
+import io.cryostat.net.security.ResourceAction;
+import io.cryostat.net.web.http.HttpMimeType;
+import io.cryostat.net.web.http.api.ApiVersion;
+import io.cryostat.net.web.http.api.v2.AbstractV2RequestHandler;
+import io.cryostat.net.web.http.api.v2.ApiException;
+import io.cryostat.net.web.http.api.v2.IntermediateResponse;
+import io.cryostat.net.web.http.api.v2.RequestParameters;
+import io.cryostat.recordings.RecordingArchiveHelper;
+import io.cryostat.recordings.RecordingNotFoundException;
+import io.cryostat.recordings.RecordingSourceTargetNotFoundException;
+
+import com.google.gson.Gson;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+
+public class RecordingGetHandler extends AbstractV2RequestHandler<Path> {
+
+    static final String PATH = "recordings/:sourceTarget/:recordingName";
+
+    private final RecordingArchiveHelper recordingArchiveHelper;
+
+    @Inject
+    RecordingGetHandler(
+            AuthManager auth, Gson gson, RecordingArchiveHelper recordingArchiveHelper) {
+        super(auth, gson);
+        this.recordingArchiveHelper = recordingArchiveHelper;
+    }
+
+    @Override
+    public boolean requiresAuthentication() {
+        return true;
+    }
+
+    @Override
+    public ApiVersion apiVersion() {
+        return ApiVersion.BETA;
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    @Override
+    public Set<ResourceAction> resourceActions() {
+        return EnumSet.of(ResourceAction.READ_RECORDING);
+    }
+
+    @Override
+    public String path() {
+        return basePath() + PATH;
+    }
+
+    @Override
+    public List<HttpMimeType> produces() {
+        return List.of(HttpMimeType.OCTET_STREAM);
+    }
+
+    @Override
+    public boolean isAsync() {
+        return false;
+    }
+
+    @Override
+    public IntermediateResponse<Path> handle(RequestParameters params) throws Exception {
+        String sourceTarget = params.getPathParams().get("sourceTarget");
+        String recordingName = params.getPathParams().get("recordingName");
+        try {
+            recordingArchiveHelper.validateSourceTarget(sourceTarget);
+            Path archivedRecording =
+                    recordingArchiveHelper.getRecordingPath(sourceTarget, recordingName).get();
+            return new IntermediateResponse<Path>()
+                    .addHeader(
+                            HttpHeaders.CONTENT_LENGTH,
```

Comment:
I think this header is already handled by the Vert.x `sendFile` call that is done [under the hood](https://github.com/cryostatio/cryostat/blob/f1ef50dc3ee941fa880242ee90e4247260c4bd9f/src/main/java/io/cryostat/net/web/http/api/v2/AbstractV2RequestHandler.java#L212).